core/src/test/java/com/expedia/aquila/core/util/MathUtilTest.java

```diff
@@ -0,0 +1,47 @@
+package com.expedia.aquila.core.util;
+
+import org.junit.rules.ExpectedException;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.junit.Assert.assertArrayEquals;
+
+public class MathUtilTest {
+
+    @Rule public final ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void incrAndLog() {
+      assertArrayEquals(new double[]{0.6931471805599453}, MathUtil.incrAndLog(new double[]{1.0}), 0.0);
+    }
+
+    @Test
+    public void exp() {
+      assertArrayEquals(new double[]{2.718281828459045}, MathUtil.exp(new double[]{1.0}), 0.0);
+    }
+
+    @Test
+    public void expAndDecr() {
+      assertArrayEquals(new double[]{1.718281828459045}, MathUtil.expAndDecr(new double[]{1.0}), 0.0);
+    }
+
+    @Test
+    public void weightedMovingAverageWindowSizeEven() {
+      thrown.expect(IllegalArgumentException.class);
+      MathUtil.weightedMovingAverage(new double[]{1.0, 2.0}, 2);
+      // Method is not expected to return due to exception thrown
+    }
+
+    @Test
+    public void weightedMovingAverageDataTooSmall() {
+      thrown.expect(IllegalArgumentException.class);
+      MathUtil.weightedMovingAverage(new double[]{}, 1);
+      // Method is not expected to return due to exception thrown
+    }
+
+    @Test
+    public void weightedMovingAverageInput0NegativeOutput0() {
```

Comment:
Change this test name to match the other styles

Reply:
Done